### DISCUSSION TITLE
Get OIDC access token on authorization code

### DISCUSSION THREAD

**Discussion by feco911** [ory/kratos#4127] [Q&A]
Hi,

I have integrated Ory Kratos with our custom company-used OIDC provider (call it IP). It works fine. For some business requirements I need to fetch the IP issued authorization code (from React frontend), which is currently received by Kratos via the callback on the successful authorization. Is there any callback which I can hook and receive the authorization code?

Can you please advice how can I achieve the above?

Thanks in advance.

**Comment by vinckr** [comment]
Hey @feco911 
Apologies for the late answer!
Ory Kratos typically handles the OIDC flow internally, including receiving and exchanging the authorization code for tokens. The authorization code is not usually exposed directly to the frontend application for security reasons.

Maybe this is something you can handle in your custom OIDC provider directly? 

Alternatively Ory also offers support for companies that rely on Ory software as business-critical. This can also include custom development for features that might be out of scope for the free community versions of Ory. You can [contact the team](https://ory.sh/contact) here if that is something you would like to explore.
